File: newsapp/src/main/java/com/example/newsapp/entity/NewsPortal.java
package com.example.newsapp.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "news_portals")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class NewsPortal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String url;
    private String description;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: infohub_app/lib/models/news_portal.dart
import 'category.dart';

class NewsPortal {
  final int? id;
  final String title;
  final String url;
  final String description;
  final Category? category;

  NewsPortal({
    this.id,
    required this.title,
    required this.url,
    required this.description,
    this.category,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'url': url,
      'description': description,
      'category': category?.toJson(),
    };
  }

  factory NewsPortal.fromJson(Map<String, dynamic> json) {
    return NewsPortal(
      id: json['id'],
      title: json['title'] ?? '',
      url: json['url'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] != null
          ? Category.fromJson(json['category'])
          : null,
    );
  }
}